src/auction.js

```diff
@@ -148,15 +149,25 @@ export function newAuction({adUnits, adUnitCodes, callback, cbTimeout, labels})
         return innerBidRequestId === bidRequest.bidderRequestId;
       });
       request.doneCbCallCount += 1;
-      if (_bidderRequests.every((bidRequest) => bidRequest.doneCbCallCount >= 1)) {
-        // when all bidders have called done callback atleast once it means auction is complete
-        utils.logInfo(`Bids Received for Auction with id: ${_auctionId}`, _bidsReceived);
-        _auctionStatus = AUCTION_COMPLETED;
-        executeCallback(false, true);
+      // In case of mediaType video and prebidCache enabled, call bidsBackHandler after cache is stored.
+      if ((request.bids.filter(videoAdUnit).length == 0) || (request.bids.filter(videoAdUnit).length > 0 && !config.getConfig('usePrebidCache'))) {
+        bidsBackAll()
       }
     }, 1);
   }
 
+  /**
+   * Execute bidBackHandler if all bidders have called done.
+   */
+  function bidsBackAll() {
+    if (_bidderRequests.every((bidRequest) => bidRequest.doneCbCallCount >= 1)) {
+      // when all bidders have called done callback atleast once it means auction is complete
+      utils.logInfo(`Bids Received for Auction with id: ${_auctionId}`, _bidsReceived);
+      _auctionStatus = AUCTION_COMPLETED;
+      executeCallback(false, true);
+    }
```

Comment:
should we emit auction end event here?

Reply:
Not here because executeCallback also gets called in case of timeout.
So emitted auction_end just before executing `bidsBackhandler`